Enable testifylint to lint test files, and mechanically fix reported issues

giorio94 opened the pull request:
Enable testifylint to lint the usages of github.com/stretchr/testify, in an effort to improve the readability of tests (and failure outputs) and reduce the chance of flakes. This also aims to remove the need for mechanical code review comments to encourage the usage of idiomatic testify primitives.
    
The current configuration disables the "float-compare", "go-require" and "require-error" checkers, as extra effort is required to fix the issues reported by them (require-error appears to also lead to false positives when used inside EventuallyWithT blocks).

Note to the reviewers, all except the first five commits, and the last one, have been mechanically generated with the following script:

```bash
#!/bin/bash

while read -r LINTER; do

[[ ${LINTER} =~ ^# ]] && continue

# Required to prevent error-nil from attempting to apply the same fix twice to
# a single code-block, failing due to conflicts.
if [[ ${LINTER} = "error-nil" ]]; then
    find . -iname '*_test.go' | \
        xargs sed -i -e 's/require.Nil(t, testutils.WaitUntil/require.NoError(t, testutils.WaitUntil/'
fi

testifylint --fix --disable-all --enable ${LINTER} ./...
git status -s | awk '{ print $2}' | xargs -r goimports -w

git add .
git commit --signoff -F- <<EOF
treewide(tests): testifylint "${LINTER}" fixes

Generated by:

    testifylint --fix --disable-all --enable ${LINTER} ./...
EOF

done <<-EOF
error-nil
bool-compare
compares
contains
empty
error-is-as
expected-actual
formatter
len
negative-positive
nil-compare
regexp
suite-broken-parallel
suite-dont-use-pkg
suite-extra-assert-call
suite-thelper
EOF
```

giorio94 committed:
api: drop unnecessary assertions from TestKafkaEqual test

Drop unnecessary assertions, as they were basically testing constants.

Signed-off-by: Marco Iorio <marco.iorio@isovalent.com>

giorio94 committed:
golangci-lint: enable testifylint

Enable testifylint to lint the usages of github.com/stretchr/testify,
in an effort to improve the readability of tests (and failure outputs)
and reduce the chance of flakes. This also aims to remove the need for
mechanical code review comments to encourage the usage of idiomatic
testify primitives.

The current configuration disables the "float-compare", "go-require"
and "require-error" checkers, as extra effort is required to fix the
issues reported by them (require-error appears to also lead to false
positives when used inside EventuallyWithT blocks).

Signed-off-by: Marco Iorio <marco.iorio@isovalent.com>

giorio94 committed:
treewide(tests): testifylint "error-nil" fixes

Generated by:

    testifylint --fix --disable-all --enable error-nil ./...

Signed-off-by: Marco Iorio <marco.iorio@isovalent.com>

giorio94 committed:
treewide(tests): testifylint "bool-compare" fixes

Generated by:

    testifylint --fix --disable-all --enable bool-compare ./...

Signed-off-by: Marco Iorio <marco.iorio@isovalent.com>

giorio94 committed:
treewide(tests): testifylint "compares" fixes

Generated by:

    testifylint --fix --disable-all --enable compares ./...

Signed-off-by: Marco Iorio <marco.iorio@isovalent.com>

giorio94 committed:
treewide(tests): testifylint "contains" fixes

Generated by:

    testifylint --fix --disable-all --enable contains ./...

Signed-off-by: Marco Iorio <marco.iorio@isovalent.com>

giorio94 committed:
treewide(tests): testifylint "empty" fixes

Generated by:

    testifylint --fix --disable-all --enable empty ./...

Signed-off-by: Marco Iorio <marco.iorio@isovalent.com>

giorio94 committed:
treewide(tests): testifylint "error-is-as" fixes

Generated by:

    testifylint --fix --disable-all --enable error-is-as ./...

Signed-off-by: Marco Iorio <marco.iorio@isovalent.com>

giorio94 committed:
treewide(tests): testifylint "expected-actual" fixes

Generated by:

    testifylint --fix --disable-all --enable expected-actual ./...

Manually restored the comments in one of the TestParseGetLabelValues
assertions, which got incorrectly deleted by the mechanical fix.

Signed-off-by: Marco Iorio <marco.iorio@isovalent.com>

giorio94 committed:
treewide(tests): testifylint "formatter" fixes

Generated by:

    testifylint --fix --disable-all --enable formatter ./...

Signed-off-by: Marco Iorio <marco.iorio@isovalent.com>

giorio94 committed:
treewide(tests): testifylint "len" fixes

Generated by:

    testifylint --fix --disable-all --enable len ./...

Signed-off-by: Marco Iorio <marco.iorio@isovalent.com>

giorio94 committed:
treewide(tests): testifylint "negative-positive" fixes

Generated by:

    testifylint --fix --disable-all --enable negative-positive ./...

Signed-off-by: Marco Iorio <marco.iorio@isovalent.com>

giorio94 committed:
treewide(tests): testifylint "nil-compare" fixes

Generated by:

    testifylint --fix --disable-all --enable nil-compare ./...

Signed-off-by: Marco Iorio <marco.iorio@isovalent.com>

giorio94 committed:
ipam: fix Test_MultiPoolManager test

In preparation for the subsequent testifylint mechanical changes, let's
convert an incorrect Error assertion into the corresponding ErrorContains
one. Error asserts that the second argument is a non-nil error, which in
this case was always true.

Signed-off-by: Marco Iorio <marco.iorio@isovalent.com>

giorio94 committed:
maglev: minor change in TestWeightedBackendWithRemoval test

In preparation for the subsequent testifylint mechanical changes, let's
anticipate these modifications. The mechanical ones would indeed lead
to a failing test due to using Equal rather than EqualValues, which also
asserts type equality (map values are of type uint64 here).

Signed-off-by: Marco Iorio <marco.iorio@isovalent.com>

giorio94 committed:
policy: minor change in TestSelectorManagerCanGetBeforeSet test

In preparation for the subsequent testifylint mechanical changes, let's
anticipate these modifications. The mechanical ones would indeed lead
to a failing test when asserting that the `selections` variable is nil,
because the zero type of a slice is not equal to nil, but satisfies
assert.Nil. Hence, let's just check that the slice is correctly empty.

For reference, all assertions in the following test pass:

    func TestFoo(t *testing.T) {
	var foo []int
	assert.NotEqual(t, nil, foo)
	assert.Nil(t, foo)
	assert.Empty(t, foo)
    }

Signed-off-by: Marco Iorio <marco.iorio@isovalent.com>

rolinh commented on pkg/hubble/metrics/api/context_test.go:
```diff
@@ -514,26 +514,24 @@ func TestParseGetLabelValues(t *testing.T) {
 	assert.EqualValues(t,
-		mustGetLabelValues(opts, flow),
 		[]string{
-			// source_ip, source_pod, source_namespace, source_workload, source_workload_kind , source_app
```
I think these comments were somewhat useful given it's just a slice of strings

giorio94 replied:
Sure, I'll restore them (it seems that `--fix` is not smart enough to preserve the comments :disappointed:.

gandro reviewed (approved):
Looks good for my codeowners (besides what Robin pointed out in one of the Hubble files)! Thanks

pippolo84 reviewed (approved):
Thanks!

derailed reviewed (approved):
@giorio94 Nice work Marco!

sayboras reviewed (approved):
Thanks a lot, a lot of linting error is probably due to my mistake.

giorio94 committed:
bandwidth: fix TestOps

In preparation for the subsequent testifylint mechanical changes, let's
fix an incorrect NoError assertion, so that it actually checks the
error returned by LinkSetUp, rather than the error already validated
by the previous assertion.

Signed-off-by: Marco Iorio <marco.iorio@isovalent.com>
